uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/apis/entitlement-api_test.js

```diff
@@ -0,0 +1,88 @@
+define([
+  'mocks/data/entitlement-data',
+  'leave-absences/shared/apis/entitlement-api',
+], function (mockData) {
+  'use strict';
+
+  describe('EntitlementAPI', function () {
+    var EntitlementAPI, $httpBackend;
+
+    beforeEach(module('leave-absences.apis'));
+
+    beforeEach(inject(function (_EntitlementAPI_, _$httpBackend_) {
+      EntitlementAPI = _EntitlementAPI_;
+      $httpBackend = _$httpBackend_;
+
+      //when the URL has this pattern
+      //civicrm/ajax/rest?action=get&entity=LeavePeriodEntitlement
+      //GET /civicrm/ajax/rest?action=get&entity=LeavePeriodEntitlement&json={"sequential":1,"api.LeavePeriodEntitlement.getremainder":{"entitlement_id":"$value.id","include_future":true}}&sequential=1
+      $httpBackend.whenGET(/action\=get&entity\=LeavePeriodEntitlement/)
+        .respond(function (method, url, data, headers, params) {
+          if ('json' in params) {
+            var jsonFromParams = JSON.parse(params.json);
+            if ('api.LeavePeriodEntitlement.getremainder' in jsonFromParams) {
+
+              return [200, mockData.all({}, true)];
+            }
+          }
+
+          return [200, mockData.all()];
+        });
+
+      ///civicrm/ajax/rest?action=getbreakdown&entity=LeavePeriodEntitlement&json={}&sequential=1
+      $httpBackend.whenGET(/action=getbreakdown&entity=LeavePeriodEntitlement/)
+        .respond(mockData.breakdown());
+
+    }));
+
+    it('has expected end points', function () {
+      expect(Object.keys(EntitlementAPI)).toEqual(['all', 'breakdown']);
+    });
+
+    describe('all() entitlements', function () {
```

Comment:
This kind of "first-level" `describe`s are for the methods, there is no need to add anything else other than the method name itself
```js
describe('all()', function () {

})
```